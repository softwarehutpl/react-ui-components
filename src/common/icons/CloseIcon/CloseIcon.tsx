import styled from 'styled-components';

export interface CloseIcon {
  height?: string;
  width?: string;
  color?: any;
  topPosition?: number;
  rightPosition?: number;
  iconColor?: string;
  visibility?: string;
}

const CloseIcon = styled.a<CloseIcon>`
  position: absolute;
  right: ${({ rightPosition }) => `${rightPosition}px`};
  top: ${({ topPosition }) => `${topPosition}px`};
  width: 24px;
  height: 24px;
  opacity: 1;
  transition: all .3s;
  cursor: pointer;
  visibility: ${({ visibility }) => visibility};
  
  &:before, &:after {
    position: absolute;
    left: 15px;
    content: ' ';
    height: ${({ height }) => (height || '12px')};
    width: ${({ width }) => (width || '2px')};
    background-color: ${({ theme, color, iconColor }) => iconColor ? iconColor : (theme.colors[color].base)};
  }
  
  &:before {
    transform: rotate(45deg);
  }
  
  &:after {
    transform: rotate(-45deg);
  }
`;

const defaultProps = {
  topPosition: 24,
  rightPosition: 24,
  visibility: 'visible',
};

CloseIcon.defaultProps = defaultProps;

export default CloseIcon;
