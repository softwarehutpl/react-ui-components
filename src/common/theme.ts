const theme: { [key: string]: { [key: string]: { [key: string]: string } } } = {
  colors: {
    primary: {
      base: '#000000',
      light: '#ffffff',
      dark: '#a8b3bd',
    },
    secondary: {
      base: '#3690e3',
      light: '#75b0e6',
      dark: '#034d91',
    },
    error: {
      base: '#f44336',
      light: '#f56c62',
      dark: '#a1140a',
    },
    warning: {
      base: '#faa72d',
      light: '#ffc36b',
      dark: '#995b00',
    },
    info: {
      base: '#7979d4',
      light: '#9797d1',
      dark: '#4c4cc7',
    },
    success: {
      base: '#4caf50',
      light: '#88c28a',
      dark: '#124f14',
    },
    disabled: {
      base: '#d9d9d9',
      light: '#f0eded',
      dark: '#8c8989',
    },
  },
  // TODO: change it in dedicated task for that (change ease-out for example and add more options)
  transitions: {
    all: {
      fast: 'all 0.5s ease-out',
      mid: 'all 2s ease-out',
      slow: 'all 4s ease-out',
    },
    background: {
      fast: 'background-color 0.5s ease-out',
      mid: 'background-color 2s ease-out',
      slow: 'background-color 4s ease-out',
    },
    fontColor: {
      fast: 'color 0.5s ease-out',
      mid: 'color 2s ease-out',
      slow: 'color 4s ease-out',
    }
  }
};

export default theme;
