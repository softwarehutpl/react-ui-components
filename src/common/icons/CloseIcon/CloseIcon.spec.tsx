import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import CloseIcon from './CloseIcon';
import theme from '../../../common/theme';

describe('CloseIcon', () => {
  const tree = renderer
  .create(
    <CloseIcon theme={theme} color="primary" />
  ).toJSON();

  it('should render corectly', () => {
    expect(tree).toMatchSnapshot();
  });
});
