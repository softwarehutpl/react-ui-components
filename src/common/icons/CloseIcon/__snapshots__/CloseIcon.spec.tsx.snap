// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseIcon should render corectly 1`] = `
<a
  className="sc-AxjAm ftmxZu"
  color="primary"
  visibility="visible"
/>
`;
